nilearn/image/image.py

```diff
@@ -563,6 +563,8 @@ def index_img(imgs, index):
      (91, 109, 91)
     """
     imgs = check_niimg_4d(imgs)
+    if hasattr(index, 'values') and hasattr(index, 'iloc'):
```

Comment:
Maybe add a comment to say that this is to duck-type a pandas dataframe.

Also, should we add pandas to our CI and add a test to test this? It would feel better from a quality point of view.
We can even skip the test if pandas is not there.

Reply:
+1.

Reply:
I made a small test here, but I'm not sure how to best integrate it into CI. Would you launch two environments, one with a more recent and another with an older version of pandas? Or did you have something else in mind?